paracelsus/graph.py

```diff
@@ -48,5 +52,62 @@ def get_graph_string(
         raise ValueError(f"Unknown Format: {format}")
     transformer = transformers[format]
 
+    # Keep only the tables which were included / not-excluded
+    include_tables, exclude_tables = resolve_include_exclude_tables(
+        include_tables=include_tables, exclude_tables=exclude_tables, all_tables=set(metadata.tables.keys())
+    )
+    check_included_table_fk(metadata=metadata, include_tables=include_tables)
+    filtered_metadata = filter_metadata(metadata=metadata, include_tables=include_tables)
+
     # Save the graph structure to string.
-    return str(transformer(metadata))
+    return str(transformer(filtered_metadata))
+
+
+def resolve_include_exclude_tables(
+    exclude_tables: Set[str],
+    include_tables: Set[str],
+    all_tables: Set[str],
+) -> tuple[Set[str], Set[str]]:
+    """Given sets of tables to include/exclude and the set of all tables, resolve the ones to include/exclude."""
+    match len(include_tables), len(exclude_tables):
+        case 0, 0:
+            return all_tables, set()
+        case 0, int():
+            return all_tables - exclude_tables, exclude_tables
+        case int(), 0:
+            if not include_tables.issubset(all_tables):
+                non_existent_tables = include_tables - all_tables
+                raise ValueError(
+                    f"Some tables to include ({non_existent_tables}) don't exist within the found tables ({all_tables})."
+                )
+            return include_tables, all_tables - include_tables
+        case _:
+            raise ValueError(
+                f"Only one or none of include_tables ({include_tables}) or exclude_tables ({exclude_tables}) can contain values."
+            )
```

Comment:
I'm finding the logic hard to follow here, and it's not really clear what the two return arguments are. Some additional comments would really help here.

Reply:
I added a more complete docstring, does this make it clearer?